Sources/FoundationInternationalization/ICU/ICU+StringConverter.swift

```diff
@@ -0,0 +1,202 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(FoundationEssentials)
+import FoundationEssentials
+#endif
+internal import _FoundationICU
+
+private extension String.Encoding {
+    var _icuConverterName: String? {
+        // TODO: Replace this with forthcoming(?) public property such as https://github.com/swiftlang/swift-foundation/pull/1243
+        switch self {
+        case .utf8: "UTF-8"
+        case .ascii: "US-ASCII"
+        case .japaneseEUC: "EUC-JP"
+        case .isoLatin1: "ISO-8859-1"
+        case .shiftJIS: "Shift_JIS"
+        case .isoLatin2: "ISO-8859-2"
+        case .unicode: "UTF-16"
+        case .windowsCP1251: "windows-1251"
+        case .windowsCP1252: "windows-1252"
+        case .windowsCP1253: "windows-1253"
+        case .windowsCP1254: "windows-1254"
+        case .windowsCP1250: "windows-1250"
+        case .iso2022JP: "ISO-2022-JP"
+        case .macOSRoman: "macintosh"
+        case .utf16BigEndian: "UTF-16BE"
+        case .utf16LittleEndian: "UTF-16LE"
+        case .utf32: "UTF-32"
+        case .utf32BigEndian: "UTF-32BE"
+        case .utf32LittleEndian: "UTF-32LE"
+        default: nil
+        }
+    }
+}
+
+extension ICU {
+    final class StringConverter: @unchecked Sendable {
```

Comment:
nit: does this need the `@unchecked`? I would have assumed that since it's a `final` class with all immutable, `Sendable` properties that the compiler can validate this conformance

Reply:
Thank you.
That's a vestige of my first implementation where `_converter` was a bare pointer...

Reply:
Sorry.
`@unchecked` is necessary still because `LockedState` is not `Sendable`...

Reply:
Ah `LockedState` is `Sendable` but it requires its stored value to be of a `Sendable` type, and I had mistakenly thought `OpaquePointer` was `Sendable` but it is indeed not `Sendable` which is why the `Sendable` conformance isn't available there. This `@unchecked Sendable` makes sense then, since despite the pointer and anything it stores never escapes the lock.